src/backend/distributed/deparser/deparse_database_stmts.c

```diff
@@ -111,44 +145,52 @@ AppendGrantOnDatabaseStmt(StringInfo buf, GrantStmt *stmt)
 }
 
 
+static bool
+AppendBasicAlterDatabaseOptions(StringInfo buf, DefElem *def, bool
+								prefixAppendedForBasicOptions, char *dbname)
+{
+	if (!prefixAppendedForBasicOptions)
+	{
+		appendStringInfo(buf, "ALTER DATABASE %s WITH", quote_identifier(dbname));
+		prefixAppendedForBasicOptions = true;
+	}
+	DefElemOptionToStatement(buf, def, alterDatabaseOptionFormats, lengthof(
+								 alterDatabaseOptionFormats));
+	return prefixAppendedForBasicOptions;
+}
+
+
 static void
-AppendDefElemConnLimit(StringInfo buf, DefElem *def)
+AppendAlterDatabaseSetTablespace(StringInfo buf, DefElem *def, char *dbname)
 {
-	appendStringInfo(buf, " CONNECTION LIMIT %ld", (long int) defGetNumeric(def));
+	appendStringInfo(buf,
+					 "ALTER DATABASE %s SET TABLESPACE %s",
+					 quote_identifier(dbname), quote_identifier(defGetString(def)));
 }
 
 
 static void
 AppendAlterDatabaseStmt(StringInfo buf, AlterDatabaseStmt *stmt)
 {
-	appendStringInfo(buf, "ALTER DATABASE %s ", quote_identifier(stmt->dbname));
-
 	if (stmt->options)
 	{
 		ListCell *cell = NULL;
-		appendStringInfo(buf, "WITH ");
+		bool prefixAppendedForBasicOptions = false;
 		foreach(cell, stmt->options)
 		{
 			DefElem *def = castNode(DefElem, lfirst(cell));
-			if (strcmp(def->defname, "is_template") == 0)
-			{
-				appendStringInfo(buf, "IS_TEMPLATE %s",
-								 quote_literal_cstr(strVal(def->arg)));
-			}
-			else if (strcmp(def->defname, "connection_limit") == 0)
-			{
-				AppendDefElemConnLimit(buf, def);
-			}
-			else if (strcmp(def->defname, "allow_connections") == 0)
+			if (strcmp(def->defname, "tablespace") == 0)
 			{
-				ereport(ERROR,
-						errmsg("ALLOW_CONNECTIONS is not supported"));
+				AppendAlterDatabaseSetTablespace(buf, def, stmt->dbname);
+				break;
 			}
 			else
 			{
-				ereport(ERROR,
-						errmsg("unrecognized ALTER DATABASE option: %s",
-							   def->defname));
+				prefixAppendedForBasicOptions = AppendBasicAlterDatabaseOptions(buf,
+																				def,
+																				prefixAppendedForBasicOptions,
+																				stmt->
+																				dbname);
```

Comment:
```suggestion
				AppendBasicAlterDatabaseOptions(buf,
																				def,
																				&prefixAppendedForBasicOptions,
																				stmt->
																				dbname);
```
I think it is better to pass it as reference, it is shorter and looks better and more readable in style.

Reply:
Fixed